### PR TITLE
chore: have more strict supported dependencies

### PR DESCRIPTION
**Pull request by mxschmitt**
Fixes https://github.com/microsoft/playwright-pytest/issues/125

### CODE REVIEW

**Comment by jcushman** [comment]
FWIW I believe the minimum pytest version is 6.2.4. With 6.2.3 the plugin fails to load:

```
$ pip install pytest==6.2.3 git+https://github.com/microsoft/playwright-pytest.git#egg=pytest-playwright
$ pytest --help
Traceback (most recent call last):
  File "/Users/jcushman/.pyenv/versions/pytest-playwright-version/bin/pytest", line 8, in <module>
    sys.exit(console_main())
  File "/Users/jcushman/.pyenv/versions/3.10.5/envs/pytest-playwright-version/lib/python3.10/site-packages/_pytest/config/__init__.py", line 185, in console_main
    code = main()
  File "/Users/jcushman/.pyenv/versions/3.10.5/envs/pytest-playwright-version/lib/python3.10/site-packages/_pytest/config/__init__.py", line 143, in main
    config = _prepareconfig(args, plugins)
  File "/Users/jcushman/.pyenv/versions/3.10.5/envs/pytest-playwright-version/lib/python3.10/site-packages/_pytest/config/__init__.py", line 318, in _prepareconfig
    config = pluginmanager.hook.pytest_cmdline_parse(
  File "/Users/jcushman/.pyenv/versions/3.10.5/envs/pytest-playwright-version/lib/python3.10/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/Users/jcushman/.pyenv/versions/3.10.5/envs/pytest-playwright-version/lib/python3.10/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/Users/jcushman/.pyenv/versions/3.10.5/envs/pytest-playwright-version/lib/python3.10/site-packages/pluggy/manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
  File "/Users/jcushman/.pyenv/versions/3.10.5/envs/pytest-playwright-version/lib/python3.10/site-packages/pluggy/callers.py", line 203, in _multicall
    gen.send(outcome)
  File "/Users/jcushman/.pyenv/versions/3.10.5/envs/pytest-playwright-version/lib/python3.10/site-packages/_pytest/helpconfig.py", line 100, in pytest_cmdline_parse
    config: Config = outcome.get_result()
  File "/Users/jcushman/.pyenv/versions/3.10.5/envs/pytest-playwright-version/lib/python3.10/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/Users/jcushman/.pyenv/versions/3.10.5/envs/pytest-playwright-version/lib/python3.10/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/Users/jcushman/.pyenv/versions/3.10.5/envs/pytest-playwright-version/lib/python3.10/site-packages/_pytest/config/__init__.py", line 1003, in pytest_cmdline_parse
    self.parse(args)
  File "/Users/jcushman/.pyenv/versions/3.10.5/envs/pytest-playwright-version/lib/python3.10/site-packages/_pytest/config/__init__.py", line 1283, in parse
    self._preparse(args, addopts=addopts)
  File "/Users/jcushman/.pyenv/versions/3.10.5/envs/pytest-playwright-version/lib/python3.10/site-packages/_pytest/config/__init__.py", line 1172, in _preparse
    self.pluginmanager.load_setuptools_entrypoints("pytest11")
  File "/Users/jcushman/.pyenv/versions/3.10.5/envs/pytest-playwright-version/lib/python3.10/site-packages/pluggy/manager.py", line 299, in load_setuptools_entrypoints
    plugin = ep.load()
  File "/Users/jcushman/.pyenv/versions/3.10.5/lib/python3.10/importlib/metadata/__init__.py", line 171, in load
    module = import_module(match.group('module'))
  File "/Users/jcushman/.pyenv/versions/3.10.5/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "/Users/jcushman/.pyenv/versions/3.10.5/envs/pytest-playwright-version/lib/python3.10/site-packages/_pytest/assertion/rewrite.py", line 161, in exec_module
    source_stat, co = _rewrite_test(fn, self.config)
  File "/Users/jcushman/.pyenv/versions/3.10.5/envs/pytest-playwright-version/lib/python3.10/site-packages/_pytest/assertion/rewrite.py", line 356, in _rewrite_test
    co = compile(tree, fn_, "exec", dont_inherit=True)
TypeError: required field "lineno" missing from alias
```

I think it also might be a best practice not to set version caps (max accepted version) in python as opposed to JS ([article](https://iscinumpy.dev/post/bound-version-constraints/#tldr)), though I haven't looked around to see if that view is consensus.

File: setup.py [hunk 1]
```diff
@@ -15,9 +15,9 @@
     include_package_data=True,
     install_requires=[
         "playwright>=1.18",
-        "pytest",
-        "pytest-base-url",
-        "python-slugify",
+        "pytest>=6.0.0,<8.0.0",
+        "pytest-base-url>=1.0.0,<3.0.0",
+        "python-slugify>=6.0.0,<7.0.0",
```
Comment (rwoll) [inline setup.py:20]: How did you decide on these version ranges?

Comment (mxschmitt) [reply setup.py:20]: Updated the Pytest version range.

The new values are the minimum which make the test-suite pass: `pip install pytest-base-url==1.0.0 python-slugify==6.0.0 pytest==6.2.0`

**Comment by mxschmitt** [comment]
@jcushman looks like Pytest is doing it as well, set a minimum and a maximum, so I think it seems fine and can be considered safe when e.g. pytest introduces a breaking change at some point: https://github.com/pytest-dev/pytest-base-url/blob/master/pyproject.toml#L44

**Review by rwoll — APPROVED** [review]

**Pull request merged**